src/SkyHope.LibraryManager.WebApi/Controllers/BookController.cs

```diff
@@ -246,5 +264,32 @@ public async Task<ActionResult<HttpBook>> GetBooksByYearAsync(int startYear, int
 
             return Ok(results);
         }
+
+        [HttpGet("user/{userId}")]
+        public async Task<ActionResult<List<HttpBook>>> BooksByUserAsync(int userId)
+        {
+            var checkedOutBooks = new List<HttpBook>();
+            var user = await _repository.FindAsync<User>(userId);
+            var books = user.CheckedOutBooks.ToList();
```

Comment:
I wouldn't use the navigation property here - you're loading the user when you don't need to. You could just use a specification to just filter the books by User Id.